File: Data Structures & Algorithms/maximum-subarray/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    maxSubArray(nums) {
        let ans = -Infinity, cur = 0;
        for(const n of nums) {
            cur = Math.max(n, cur + n);
            ans = Math.max(ans, cur);
        }
        return ans;
    }
}

File: Data Structures & Algorithms/merge-intervals/submission-3.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort()
        ans = []
        for st, ed in intervals:
            if not ans or ans[-1][1] < st:
                ans.append([st, ed])
            else:
                ans[-1][1] = max(ans[-1][1], ed)
        return ans